backends/p4tools/common/core/target.cpp

```diff
@@ -6,15 +6,14 @@
 #include <map>
 #include <optional>
 #include <string>
-#include <utility>
 
 #include "backends/p4tools/common/lib/variables.h"
 #include "ir/irutils.h"
 
 namespace P4::P4Tools {
 
-Target::Spec::Spec(std::string deviceName, std::string archName)
-    : deviceName(std::move(deviceName)), archName(std::move(archName)) {
+Target::Spec::Spec(std::string_view deviceName, std::string_view archName)
```

Comment:
Not sure about string_view here, can revert if necessary.

Reply:
probably OK, and the original solution was OK too. The original is a bit better when an expression retuning `std::string` is used, this one is better with non-string inputs (e.g. constants). But it does not matter much anyway, the string will be short.